src/api/OnmsAuthConfig.ts

```diff
@@ -4,20 +4,29 @@
  */
 export class OnmsAuthConfig {
   /** The password to authenticate with. */
-  public password: string;
+  public password?: string;
 
   /** The username to connect as. */
-  public username: string;
+  public username?: string;
 
   /**
    * Construct an auth configuration object.
    * @constructor
    */
-  constructor(username: string, password: string) {
+  constructor(username?: string, password?: string) {
```

Comment:
Does it make sense to make this optional?

Reply:
This actually is coming from existing use of the API that is creating an empty `OnmsAuthConfig` object and then filling in the data later.  It's not pretty, but fixing it will require a big rework of some other things and it's too big of a change to do right now.

Reply:
Oops, I commented  as bamboo rather than me.  ;)